src/test/java/seedu/address/logic/commands/FindAppCommandTest.java

```diff
@@ -0,0 +1,183 @@
+                        Optional.of(new Date("2024-12-30")),
+                        Optional.of(new Time("09:00"))
+                );
+
+        FindAppCommand findFirstCommand = new FindAppCommand(firstPredicate);
+        FindAppCommand findSecondCommand = new FindAppCommand(secondPredicate);
+
+        // same object -> returns true
+        assertTrue(findFirstCommand.equals(findFirstCommand));
+
+        // same values -> returns true
+        FindAppCommand findFirstCommandCopy = new FindAppCommand(firstPredicate);
+        assertTrue(findFirstCommand.equals(findFirstCommandCopy));
+
+        // different types -> returns false
+        assertFalse(findFirstCommand.equals(1));
+
+        // null -> returns false
+        assertFalse(findFirstCommand.equals(null));
+
+        // different person -> returns false
+        assertFalse(findFirstCommand.equals(findSecondCommand));
+    }
+
+    @Test
+    public void execute_zeroKeywords_invalidCommandFormat() {
+        assertThrows(ParseException.class, () -> preparePredicate(""));
+    }
+
+    @Test
+    public void execute_multipleKeywords_oneAppointmentFound() throws ParseException {
+        String expectedMessage = String.format(MESSAGE_APPOINTMENTS_LISTED_OVERVIEW, 1);
+        AppointmentContainsKeywordsPredicate predicate = preparePredicate(
+                " i/T0123456A d/2024-03-16 from/11:00"
+        );
+        FindAppCommand command = new FindAppCommand(predicate);
+        expectedModel.updateFilteredAppointmentList(predicate);
+        assertCommandSuccess(command, model, expectedMessage, expectedModel);
+        assertEquals(Arrays.asList(AMY_APPT), model.getFilteredAppointmentList());
+    }
+
+    @Test
+    public void execute_sameNric_twoAppointmentsFound() throws ParseException {
+        String expectedMessage = String.format(MESSAGE_APPOINTMENTS_LISTED_OVERVIEW, 3);
+        AppointmentContainsKeywordsPredicate predicate = preparePredicate(
+                " i/T0123456A"
+        );
+        FindAppCommand command = new FindAppCommand(predicate);
+        expectedModel.updateFilteredAppointmentList(predicate);
+        assertCommandSuccess(command, model, expectedMessage, expectedModel);
+        assertEquals(Arrays.asList(ALICE_APPT, ALICE_APPT_1, AMY_APPT), model.getFilteredAppointmentList());
+    }
+
+    @Test
+    public void execute_sameNricAndTime_twoAppointmentsFound() throws ParseException {
+        String expectedMessage = String.format(MESSAGE_APPOINTMENTS_LISTED_OVERVIEW, 2);
+        AppointmentContainsKeywordsPredicate predicate = preparePredicate(
+                " i/T0123456A from/16:00"
+        );
+        FindAppCommand command = new FindAppCommand(predicate);
+        expectedModel.updateFilteredAppointmentList(predicate);
+        assertCommandSuccess(command, model, expectedMessage, expectedModel);
+        assertEquals(Arrays.asList(ALICE_APPT, ALICE_APPT_1), model.getFilteredAppointmentList());
+    }
+
```

Comment:
Perhaps we can add more test cases with invalid fields?

Reply:
There are some tests in FindAppCommandParserTest.java, would those suffice?